src/main/java/core/basesyntax/dao/DBReaderImpl.java

```
@@ -0,0 +1,36 @@
+package core.basesyntax.dao;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DBReaderImpl implements DBReader {
+    File fileName;
+    FileReader reader;
+    BufferedReader bf;
+
+    public DBReaderImpl(String fileName) {
+        File file = new File(fileName);
+        try {
+            reader = new FileReader(file);
+            bf = new BufferedReader(reader);
+        } catch(FileNotFoundException e) {
+            throw new RuntimeException("Can't open file " + fileName, e);}
+    }
+
+    @Override
+    public List<String> read() {
+        List<String> list = new ArrayList<>();
+        try {
+            String header = bf.readLine();
+            String line = bf.readLine();
+            while (line != null) {
+                list.add(line);
+                line = bf.readLine();
+            }
+        } catch (IOException e) {
+            System.out.println("Error while reading file " + fileName);
```

This violates checklist item #6: 'Don't ignore exceptions in try-catch blocks'. Instead of printing to the console, you should rethrow a `RuntimeException` with an informative message and the original exception as the cause. For example: `throw new RuntimeException("Error while reading file " + fileName, e);`. Additionally, the `fileName` field used here is never initialized, so the error message would be incomplete.